src/transformers/generation/candidate_generator.py

```diff
@@ -732,15 +735,17 @@ def get_candidates(self, input_ids: torch.LongTensor) -> Tuple[torch.LongTensor,
         generation_args["generation_config"].return_dict_in_generate = True
 
         # Generate and process outputs using translator
+        generation_args['logits_processor'] = self._atm_translator.logits_processors
         assistant_output = self.assistant_model.generate(**generation_args, **self.assistant_kwargs)
         self.assistant_kwargs["past_key_values"] = assistant_output.past_key_values
 
         candidate_logits = torch.stack(assistant_output.scores, dim=1)
 
         # Use translator to convert tokens and logits
         candidate_ids = assistant_output.sequences
-        candidate_logits = self._atm_translator.logits_processors(input_ids=candidate_ids, scores=candidate_logits)
+        self._prev_assistant_ids = candidate_ids
```

Comment:
Wouldn't it be easier to set `self._prev_assistant_ids` to be the output of `self._prepare_assistant_input_ids(target_input_ids)` so that the function `_prepare_assistant_input_ids` doesn't have to crop out its suffix?

Reply:
`self._prev_assistant_ids` includes also the candidates not yet validated while in `self._prepare_assistant_input_ids` we crop unvalidated draft tokens.

Reply:
Okay. Perhaps we can simplify this in later PRs.